src/rules/no-entire-object-hook-deps.ts

```
@@ -202,17 +202,36 @@ function getObjectUsagesInHook(
             current.type === AST_NODE_TYPES.Identifier &&
             current.name === objectName
           ) {
-            // This is an array/string method on our target object, so we need the entire object
-            needsEntireObject = true;
+            if (pathParts.length === 0) {
+              // Direct method call on the object (e.g., userData.map(...))
+              needsEntireObject = true;
+            } else {
+              // Method call on a property (e.g., userData.items.map(...) or userData?.items?.map(...))
+              let path = objectName + (hasOptionalChaining ? '?' : '');
+              path += '.' + pathParts.join('.');
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)